converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -36,7 +37,7 @@ func Convert(in []byte) ([]byte, diag.Diagnostics) {
 	}
 
 	f := builder.NewFile()
-	diags = AppendAll(f, promConfig, "")
+	diags = AppendAll(f, promConfig, "", []discovery.Target{}, nil)
```

Comment:
Maybe we can have a couple of specialised `Append*` - which would also work better with the suggestion I made below regarding the label function passed as an argument?

For example:
* `AppendAll(file, prom_config)` -> calls `AppendAllNested(....)` with some hard-coded arguments
* `AppendAllNested(file, prom_config, jobNameToCompLabelsFunc, extraTargets, remoteWriteExports)`